File: Backend/LinkCare/src/main/java/com/ssafy/linkcare/group/dto/GroupDetailResponse.java
package com.ssafy.linkcare.group.dto;

import com.ssafy.linkcare.group.enums.GroupType;

import java.time.LocalDateTime;
import java.util.List;

/*
    * 그룹 상세 정보 응답 DTO
        * - 그룹 상세 조회 시 사용
        * - 멤버 목록, 목표 기준 포함
*/
public record GroupDetailResponse(
        Long groupSeq,
        String groupName,
        String groupDescription,
        GroupType type,
        Integer capacity,
        Integer currentMembers,  // 현재 인원 수
        String imageUrl,
        LocalDateTime createdAt,
        GoalCriteriaDto goalCriteria,  // HEALTH 그룹인 경우에만 존재 (nullable)
        List<MemberDto> members,
        Long currentUserSeq  // 현재 로그인한 사용자의 userSeq
) {
    // 멤버 정보 (Nested Record)
    public record MemberDto(
            Long groupMemberSeq,
            Long userSeq,
            String userName,
            Boolean isLeader,
            String mainCharacterBaseImageUrl
    ) {}
}

File: Frontend/app/src/main/java/com/a307/linkcare/feature/commongroup/data/model/response/GroupDetailResponse.kt
package com.a307.linkcare.feature.commongroup.data.model.response

import com.google.gson.annotations.SerializedName

data class GroupDetailResponse(
    val groupSeq: Long,
    val groupName: String,
    val groupDescription: String,
    val type: String,
    val capacity: Int,
    val currentMembers: Int,
    val imageUrl: String,
    val createdAt: String,
    val goalCriteria: GoalCriteria?,
    val members: List<GroupMemberResponse>,
    val currentUserSeq: Long  // 현재 로그인한 사용자의 userSeq
)

data class GoalCriteria(
    val minCalorie: Double,
    val minStep: Int,
    val minDistance: Double,
    val minDuration: Int
)

data class GroupMemberResponse(
    val groupMemberSeq: Long,
    val userSeq: Long,
    val userName: String,
    val isLeader: Boolean,
    val mainCharacterBaseImageUrl: String
)

data class GoalCriteriaDto(
    @SerializedName("isSleepAllowed") val isSleepAllowed: Boolean?,
    @SerializedName("isWaterIntakeAllowed") val isWaterIntakeAllowed: Boolean?,
    @SerializedName("isBloodPressureAllowed") val isBloodPressureAllowed: Boolean?,
    @SerializedName("isBloodSugarAllowed") val isBloodSugarAllowed: Boolean?
)
